python/pyarrow/interchange/column.py

```diff
@@ -312,7 +312,12 @@ def _dtype_from_arrowdtype(
             return kind, bit_width, f_string, Endianness.NATIVE
         elif pa.types.is_dictionary(dtype):
             kind = DtypeKind.CATEGORICAL
-            f_string = "L"
+            arr = self._col
+            indices_dtype = arr.indices.type
+            _, f_string = _PYARROW_KINDS.get(indices_dtype, (None, None))
+            if f_string is None:
+                raise ValueError(
+                    f"Data type {indices_dtype} not supported by interchange protocol")
```

Comment:
In theory this should never happen, right? (because we know that our indices types are supported by the protocol and are inside `_PYARROW_KINDS`) Then it could also be simplified a bit.

Reply:
Yeah, that was my thought but was not sure 😊 But here it is:
https://github.com/apache/arrow/blob/76f987eb94d2832bc959de455847a03387cab41a/python/pyarrow/types.pxi#L4486-L4489

Will remove 👍